openml/extensions/sklearn/extension.py

```diff
@@ -0,0 +1,318 @@
+        """Instantiate a scikit-learn model from the flow representation.
+
+        Parameters
+        ----------
+        flow : OpenMLFlow
+
+        Returns
+        -------
+        Any
+        """
+        return flow_to_sklearn(flow)
+
+    def model_to_flow(self, model: Any) -> 'OpenMLFlow':
+        """Transform a scikit-learn model to a flow for uploading it to OpenML.
+
+        Parameters
+        ----------
+        model : Any
+
+        Returns
+        -------
+        OpenMLFlow
+        """
+        return sklearn_to_flow(model)
+
+    def get_version_information(self) -> List[str]:
+        """List versions of libraries required by the flow.
+
+        Libraries listed are ``Python``, ``scikit-learn``, ``numpy`` and ``scipy``.
+
+        Returns
+        -------
+        List
+        """
+        return get_version_information()
+
+    def create_setup_string(self, model: Any) -> str:
+        """Create a string which can be used to reinstantiate the given model.
+
+        Parameters
+        ----------
+        model : Any
+
+        Returns
+        -------
+        str
+        """
+        return create_setup_string(model)
+
+    ################################################################################################
+    # Methods for performing runs with extension modules
+
+    def is_estimator(self, model: Any) -> bool:
+        """Check whether the given model is a scikit-learn estimator.
+
+        This function is only required for backwards compatibility and will be removed in the
+        near future.
+
+        Parameters
+        ----------
+        model : Any
+
+        Returns
+        -------
+        bool
+        """
+        return is_estimator(model)
+
+    def seed_model(self, model: Any, seed: Optional[int] = None) -> Any:
+        """Set the random state of all the unseeded components of a model and return the seeded
+        model.
+
+        Required so that all seed information can be uploaded to OpenML for reproducible results.
+
+        Models that are already seeded will maintain the seed. In this case,
+        only integer seeds are allowed (An exception is raised when a RandomState was used as
+        seed).
+
+        Parameters
+        ----------
+        model : sklearn model
+            The model to be seeded
+        seed : int
+            The seed to initialize the RandomState with. Unseeded subcomponents
+            will be seeded with a random number from the RandomState.
+
+        Returns
+        -------
+        Any
+        """
+        return seed_model(model, seed)
+
+    def run_model_on_fold(
+        self,
+        model: Any,
+        task: 'OpenMLTask',
+        rep_no: int,
+        fold_no: int,
+        sample_no: int,
+        add_local_measures: bool,
+    ) -> Tuple[List[List], List[List], 'OrderedDict[str, float]', Any]:
+        """Run a model on a repeat,fold,subsample triplet of the task and return prediction
```

Comment:
I understand this function signature is legacy. That said, seeing this again got me thinking - should we consider replacing parameters `task, rep_no, fold_no, sample_no` with `X_train, X_test, y_train=None`? Which would allow anyone writing an extension minimal knowledge of the intricacies of fetching the right data. It would centralize [this code](https://github.com/openml/openml-python/blob/extensions/openml/extensions/sklearn/run_functions.py#L182), so people don't need to keep re-implementing it. I might very easily be missing something, but at first glance this seems cleaner?

Reply:
Oh, I noticed the task is also needed for creating the arff content. I guess there's not much advantage then (still might be nicer if it is `task, X_train, X_test, y_train=None`... but now I'm not so sure anymore :)

Reply:
Please check my answer to [Jan's comment above](https://github.com/openml/openml-python/pull/647#discussion_r269159706). I suggested to do this in a new PR because it will require major refactoring again.